docs/Developer-guidlines.md

```diff
@@ -0,0 +1,111 @@
+# Developer Guidelines
+
+This document describes the developer policies and procedures for maintaining
+Bare Metal Imaging Service (BMI), including the accepted [coding style](#Coding-style).
+
+## Getting started
+
+If you are new to BMI and would like to jump in, then welcome!
+
+Contributions can come in many forms: installing BMI and using it in your
+environment (and giving us feedback), improving the testing, squashing bugs and
+introducing new functionality.
+
+If you are interested in any of this, a good place to start is by reading the
+documentation (linked from the [Readme](./README.html)).
+
+If you're looking to code, there are a few ways to help:
+* Improving the BMI documentation
+* Better testing
+* Reviewing [Pull Requests](https://github.com/CCI-MOC/ims/pulls)
+* Discussions on open github issues. 
+
+## Communicating
+
+* IRC: The MOC team hangs out on #moc on [freenode](https://www.freenode.net/)
+* IRL (In Real Life): The MOC group office is on the BU campus, at 3 Cummington Mall, Boston, MA room 451. Anyone interested in BMI is welcome to drop in and work there.
+* Slack: Our slack channel is at team-bmis.slack.com
+
+## Coding style/conventions
+
+By default, BMI (like many other python projects) uses
+[PEP8](https://www.python.org/dev/peps/pep-0008/) as its naming guide, and
+[PEP257](https://www.python.org/dev/peps/pep-0257/) for documentation.
+Departures are acceptable when called for, but should be discussed first.
+
+
+## Submitting code / Pull Requests
+
+Overall, BMI follows the [github fork & pull
+model](http://scottchacon.com/2011/08/31/github-flow.html) to integrate
+contributions, where users fork the main BMI (ims) repo, push changes
+to their personal fork and then create a Pull Request to merge it
+into the master branch of the BMI repository.
+
+### Prior to the pull request
+This summarizes what should be done prior to a pull request:
+
+- [ ] If functionality could have architectural implications or controversial, have a discussion with the team. Ideally, prior to coding to save effort.
+- [ ] Ensure any user, deployer or developer documentation is updated.
+- [ ] If a change affects an external API, be sure to update docs/rest\_api.md.
+- [ ] Testing:
+  - [ ] Ensure tests pass (coming soon!) after making your changes by running `py.test tests/unit ` from the top-level ims directory. Parallel testing can be used on multi-core systems by running `py.test tests/unit -n auto`
+  - [ ] Add unit tests in the corresponding file and create one if none are present.
+  - [ ] If practical, bug fixes should have an reproducing test to ensure that the bug does not come back.
+  - [ ] Run deployment tests if needed
+
+#### Get agreement
```

Comment:
I'll change it to "Get agreeement from the BMI team" (or if you have any other suggestion). Point being, before a developer starts coding, the team should, at least, agree with the idea to save coding effort.